.github/workflows/build.yml

```diff
@@ -92,6 +98,8 @@ jobs:
           RUSTFLAGS="-C link-dead-code" cargo test --verbose --color always --features rpc-client
           RUSTFLAGS="-C link-dead-code" cargo test --verbose --color always --features rpc-client,rest-client
           RUSTFLAGS="-C link-dead-code" cargo test --verbose --color always --features rpc-client,rest-client,tokio
+          cd ../lightning
+          RUSTFLAGS="-C link-dead-code" cargo test --verbose --color always --features hashbrown
```

Comment:
I dunno if we want the last test we run in "matrix.coverage" (ie the binaries which we generate coverage reports from) to be hashbrown ones - maybe move this up above (or as the first command in) "Test on Rust ${{ matrix.toolchain }} with net-tokio and full code-linking for coverage generation"?

Reply:
SGTM, wasn't sure, so just added it after the other tests.

Reply:
hrm, seems like this causes build failure:

```
error[E0658]: use of unstable library feature 'ptr_cast'
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.11.2/src/raw/mod.rs:504:57
    |
504 |         NonNull::new_unchecked(self.table.ctrl.as_ptr().cast())
    |                                                         ^^^^
    |
    = note: for more information, see https://github.com/rust-lang/rust/issues/60602
```

Reply:
Might only make sense to use `hashbrown` in later versions of Rust, per [rust-lang/rust#60602](https://github.com/rust-lang/rust/issues/60602#issuecomment-594005876)

Reply:
Right, it needs to only run on non-1.36 versions of rust. Maybe add this:
```
      - name: Test no_std builds Rust ${{ matrix.toolchain }}
        if: "matrix.build-net-tokio"
        run: cargo test --verbose --color always --features hashbrown
 ```

Reply:
> MSRV of hashbrown 0.11.2 is rust 1.49

Oh, I missed this comment, I guess we need the if there to just be something new that matches on stable instaed.

Reply:
Try:
```
diff --git a/.github/workflows/build.yml b/.github/workflows/build.yml
index 231db705..2f14a4c2 100644
--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -14,18 +14,27 @@ jobs:
                      # 1.41.0 is Debian stable
                      1.41.0,
                      # 1.45.2 is MSRV for lightning-net-tokio, lightning-block-sync, and coverage generation
-                     1.45.2]
+                     1.45.2,
+                     # 1.49 is MSRV for no_std
+                     1.49.0]
         include:
           - toolchain: stable
             build-net-tokio: true
+            build-no-std: true
           - toolchain: stable
             platform: macos-latest
             build-net-tokio: true
+            build-no-std: true
           - toolchain: stable
             platform: windows-latest
             build-net-tokio: true
+            build-no-std: true
           - toolchain: beta
             build-net-tokio: true
+            build-no-std: true
+          - toolchain: 1.45.2
+            build-net-tokio: true
+            build-no-std: true
           - toolchain: 1.45.2
             build-net-tokio: true
             coverage: true
@@ -76,6 +85,9 @@ jobs:
           cd ../lightning
           RUSTFLAGS="-C link-dead-code" cargo build --verbose --color always --features hashbrown
           cd ..
+      - name: Test no_std builds Rust ${{ matrix.toolchain }}
+        if: "matrix.build-no-std"
+        run: cargo test --verbose --color always --features hashbrown
       - name: Test on Rust ${{ matrix.toolchain }} with net-tokio
         if: "matrix.build-net-tokio && !matrix.coverage"
         run: cargo test --verbose --color always
```

Reply:
I added a check `build-no-std`, and a `1.49.0` to the toolchain. It took a little experimentation, but I think I understand the CI build scripts better now.

The fuzz build seems to be failing for other reasons...